monkey/monkey_island/cc/server_utils/encryptor.py

```diff
@@ -21,6 +23,7 @@ def __init__(self, password_file_dir):
         if os.path.exists(password_file):
             self._load_existing_key(password_file)
         else:
+            create_secure_file(path=password_file)
```

Comment:
I believe if `create_secure_file()` returns the file handler without closing, you could do something like:

```python
with open(create_secure_file(password_file), "wb"):
```

This would avoid the `os.close(os.open(...))` ugliness in the `create_secure_file()` function. You should probably then rename it to `open_new_secure_file()` or something.

Reply:
What purpose would that serve? We would just be doing something like
```python3
with open(create_secure_file(password_file), "wb"):
    pass
```
?

Plus, I don't mind the `os.close(os.open(...))` ugliness if it means that we're maintaining consistency in `utils.py` (where we're simply *creating* secure files/directories, not opening them).

Reply:
There are better ways to improve readability. Creating and opening and then closing the file is not a very intuitive workflow just for creating a file.

Reply:
You wouldn't use `open()` and then pass. The code that used it would look like this:
```diff
diff --git a/monkey/monkey_island/cc/server_utils/encryptor.py b/monkey/monkey_island/cc/server_utils/encryptor.py
index abeb34dc..9ec3dc37 100644
--- a/monkey/monkey_island/cc/server_utils/encryptor.py
+++ b/monkey/monkey_island/cc/server_utils/encryptor.py
@@ -6,7 +6,7 @@ import os
 from Crypto import Random  # noqa: DUO133  # nosec: B413
 from Crypto.Cipher import AES  # noqa: DUO133  # nosec: B413
 
-from monkey_island.cc.server_utils.file_utils import create_secure_file
+from monkey_island.cc.server_utils.file_utils import open_new_secure_file
 
 __author__ = "itay.mizeretz"
 
@@ -23,12 +23,11 @@ class Encryptor:
         if os.path.exists(password_file):
             self._load_existing_key(password_file)
         else:
-            create_secure_file(path=password_file)
             self._init_key(password_file)
 
     def _init_key(self, password_file):
         self._cipher_key = Random.new().read(self._BLOCK_SIZE)
-        with open(password_file, "wb") as f:
+        with open(open_new_secure_file(password_file, "wb")) as f:
             f.write(self._cipher_key)
 
     def _load_existing_key(self, password_file):
```

Rather than `close(open())` and then `open()` the file again later, you would just open the file and write to it. Open a file and write to it is the standard workflow for handling files. Create a directory and use it is the standard workflow for directories. There's not really an inconsistency.